rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/RMRestInterface.java

```diff
@@ -1021,4 +1030,21 @@ void removeNodeToken(@HeaderParam("sessionid") String sessionId, @HeaderParam("n
     void setNodeTokens(@HeaderParam("sessionid") String sessionId, @HeaderParam("nodeurl") String nodeUrl,
             @QueryParam("tokens") List<String> tokens) throws NotConnectedException, RestException;
 
+    @GET
+    @Path("node/tags")
+    @Produces("application/json")
+    Set<String> getNodeTags(@HeaderParam("sessionid") String sessionId, @QueryParam("nodeurl") String url)
+            throws NotConnectedException, RestException;
+
+    /**
+     * Search the nodes with specific tag.
+     * @param sessionId current session
+     * @param tag the tag which the return nodes should contains, when it's not specified, all the known nodes urls are returned
+     * @return the set of urls which match the search condition
+     */
+    @GET
+    @Path("nodes/search")
+    @Produces("application/json")
+    Set<String> searchNodes(@HeaderParam("sessionid") String sessionId, @QueryParam("tag") String tag)
```

Comment:
I think we should be able to search with multiple tags. Accordinglinly the parameter should be a list

Reply:
Finally, could you add an endpoint which returns a set of all tags present in all nodes ?

Path("nodes/tags/all")

Accordingly, it's vbetter to define the search method as
Path("nodes/tags/search")